Assets/EngyneCreations/Trello/Scripts/TrelloAPI.cs

```diff
@@ -0,0 +1,313 @@
+				CatchException("Invalid Trello Card, unable to upload");
+			}
+			
+			WWWForm post = new WWWForm();
+			post.AddField("name", card.name);
+			post.AddField("desc", card.desc);
+			post.AddField("due", card.due);
+			post.AddField("idList", card.listId);
+			post.AddField("urlSource", card.urlSource);
+			if (card.attachment.IsValid())
+			{
+				TrelloCard.Attachment attachment = card.attachment;
+				post.AddBinaryData("fileSource", attachment.FileSource, attachment.FileName);
+			}
+
+			using (UnityWebRequest request = UnityWebRequest.Post(string.Format("{0}?key={1}&token={2}", CARD_BASE_URL, KEY, TOKEN), post))
+			{
+				await request.SendWebRequest();
+
+				if (request.result == UnityWebRequest.Result.Success)
+				{
+					Debug.Log($"Trello Card \"{card.name}\" was successfully uploaded");
+				}
+				else
+				{
+					CatchException($"Unable to upload Trello Card: {request.error}");
+				}
+			}
+		}
+		
+		/// <summary>
+		/// Download the list of available boards for the user and store them.
+		/// </summary>
+		/// <returns>Downloaded boards.</returns>
+		private async Task PopulateBoards()
+		{
+			using (UnityWebRequest request = UnityWebRequest.Get(string.Format("{0}?key={1}&token={2}&boards=all", MEMBER_BASE_URL, KEY, TOKEN)))
+			{
+				await request.SendWebRequest();
+
+				_boards = null;
+				if (request.result == UnityWebRequest.Result.Success && request.downloadHandler != null)
+				{
+					var dict = Json.Deserialize(request.downloadHandler.text) as Dictionary<string,object>;
+					_boards = (List<object>)dict["boards"];
+				}
+				else
+				{
+					CatchException($"Unable to populate board from Trello : {request.error}");
+				}
+			}
+		}
+		
+		/// <summary>
+		/// Download all the lists of the selected board and store them.
+		/// </summary>
+		/// <returns>Downloaded list.</returns>
+		private async Task PopulateLists()
+		{
+			_lists = null;
+			
+			if (_currentBoardId == null)
+			{
+				CatchException("Cannot retrieve the lists, there isn't a selected board yet.");
+			}
+
+			using (UnityWebRequest request = UnityWebRequest.Get(string.Format("{0}{1}?key={2}&token={3}&lists=all", BOARD_BASE_URL, _currentBoardId, KEY, TOKEN)))
+			{
+				await request.SendWebRequest();
+
+				if (request.result == UnityWebRequest.Result.Success && request.downloadHandler != null)
+				{
+					var dict = Json.Deserialize(request.downloadHandler.text) as Dictionary<string,object>;
+					_lists = (List<object>)dict["lists"];
+				}
+				else
+				{
+					CatchException($"Unable to retrieve the lists : {request.error}");
+				}
+			}
+		}
+
+		private async Task CreateNewBoard(string boardName)
+		{
+			if (string.IsNullOrEmpty(boardName))
+			{
+				CatchException("Unable to create the new board");
+			}
+			
+			WWWForm post = new WWWForm();
+			post.AddField("name",boardName);
+
+			using (UnityWebRequest request = UnityWebRequest.Post(string.Format(string.Format("{0}?key={1}&token={2}", BOARD_BASE_URL, KEY, TOKEN)), post))
```

Comment:
You need to make this much clearer as to what's happening.

Reply:
do you mean like renaming the variable to something like `createNewTrelloBoardRequest`?

Reply:
mostly this part
`(string.Format(string.Format("{0}?key={1}&token={2}", BOARD_BASE_URL, KEY, TOKEN)), post)`

too many nesting